src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -375,6 +381,37 @@ public int compareOffsets(final NakadiCursor first, final NakadiCursor second) t
         return KafkaCursor.fromNakadiCursor(first).compareTo(KafkaCursor.fromNakadiCursor(second));
     }
 
+    public long totalEventsInPartition(final Timeline timeline, final String partitionString)
+            throws InvalidCursorOperation {
+        try {
+            final Timeline.StoragePosition positions = timeline.getLatestPosition();
+            if (positions == null) {
+                final Optional<PartitionStatistics> statsO = loadPartitionStatistics(timeline, partitionString);
```

Comment:
If we expect that there will be a lot of calls to this method, then maybe it makes sense to use KafkaConsumer here, because it will eliminate the need of seekToBegining()